checker/src/org/checkerframework/checker/index/qual/SearchIndex.java

```diff
@@ -0,0 +1,31 @@
+package org.checkerframework.checker.index.qual;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+import org.checkerframework.framework.qual.JavaExpression;
+import org.checkerframework.framework.qual.SubtypeOf;
+
+/**
+ * The annotated expression evaluates to an integer whose between -a.length - 1 and a.length - 1,
+ * inclusive, for all sequences a listed in the annotation.
+ *
+ * <p>This type is designed with one goal: to handle the convention of {@link
```

Comment:
I would just say, "binary search returns an integer of this type", without the other caveats.